common/models/user.js

```diff
@@ -756,10 +756,20 @@ module.exports = function(User) {
 
     UserModel.validatesFormatOf('email', { with: re, message: g.f('Must provide a valid email') });
 
-    // FIXME: We need to add support for uniqueness of composite keys in juggler
+    // Non-realm users validation
     if (!(UserModel.settings.realmRequired || UserModel.settings.realmDelimiter)) {
       UserModel.validatesUniquenessOf('email', { message: 'Email already exists' });
       UserModel.validatesUniquenessOf('username', { message: 'User already exists' });
+    } else {
+      // Realm users validation
```

Comment:
Can we please swap if-else blocks for easier readability?
i.e.
```javascript
if(realmRequired){
  // Realm users validation
} else {
  // Non-realm users validation
}
```

Just an opinion, leave it upto you to change or not, @davidcheung